x-pack/plugins/actions/server/builtin_action_types/lib/axios_utils.ts

```diff
@@ -32,15 +32,17 @@ export const request = async <T = unknown>({
   validateStatus?: (status: number) => boolean;
   auth?: AxiosBasicCredentials;
 }): Promise<AxiosResponse> => {
+  const { httpAgent, httpsAgent } = getProxyAgents(proxySettings, logger);
+
   return await axios(url, {
     method,
     data: data ?? {},
     params,
     auth,
-    // use httpsAgent and embedded proxy: false, to be able to handle fail on invalid certs
-    httpsAgent: proxySettings ? getProxyAgent(proxySettings, logger) : undefined,
-    httpAgent: proxySettings ? getProxyAgent(proxySettings, logger) : undefined,
```

Comment:
I cannot remember the reason why I decided to set httpsAgent and httpAgent with the same value, just only based on the user setting. Probably I wasn't able to test https version well and during manual testing manipulation did this solution - I was sure the initial idea was exactly the same as you did now :-)

Reply:
>Probably I wasn't able to test https version well and during manual testing manipulation did this solution - I was sure the initial idea was exactly the same as you did now :-)

One trick I did for this was to use Kibana as the endpoint with `--ssl`. I was then able to test an https endpoint.